app/services/withdraw_manual_service.rb

```diff
@@ -0,0 +1,38 @@
+class WithdrawManualService
+  def initialize(manual_repository:, listeners: [], manual_id:)
+    @manual_repository = manual_repository
+    @listeners = listeners
+    @manual_id = manual_id
+  end
+
+  def call
+    withdraw
+    persist
+    notify_listeners
+
+    manual
+  end
+
+private
+  attr_reader :manual_repository, :listeners, :manual_id
+
+  def withdraw
+    manual.withdraw!
+  end
+
+  def persist
+    manual_repository.store(manual)
+  end
+
+  def notify_listeners
+    listeners.each { |l| l.call(manual) }
```

Comment:
It's worth being aware that our Ruby [style guide](https://github.com/alphagov/styleguides/blob/9884028a035afb693e4c9111394e0a9f57a685e0/ruby.md#syntax) says:

> Use single-letter variables with extreme caution. These are sometimes ok in single-line blocks when the line itself explains what the variable is for, but never in multi-line blocks or where the context isn't entirely clear. If in any doubt, give your variables descriptive names.

I'm totally happy with it in this case - it's very clear to me what's happening here.